src/script/conversation/ConversationRepository.ts

```diff
@@ -2924,12 +2918,12 @@ export class ConversationRepository {
         if (!eventFromStream) {
           amplify.publish(WebAppEvents.NOTIFICATION.NOTIFY, messageEntity, undefined, conversationEntity);
         }
-
-        if (conversationEntity.is_cleared()) {
-          conversationEntity.cleared_timestamp(0);
-        }
       }
     }
+
+    if (conversationEntity.is_cleared()) {
+      conversationEntity.cleared_timestamp(0);
+    }
```

Comment:
Every we receive any conversation in the conversation we reset cleared_timestamp to 0 so we allow the end user to clear it again.